services/admin/backend/service/project/project.go

```diff
@@ -35,16 +35,34 @@ func NewService(r Repository) *Service {
 }
 
 //create new project
-func (s *Service) CreateProject(ctx context.Context, shortCode valueobject.ShortCode, shortName valueobject.ShortName, longName valueobject.LongName, description valueobject.Description) (valueobject.Identifier, error) {
+func (s *Service) CreateProject(ctx context.Context, shortName valueobject.ShortName, longName valueobject.LongName, description valueobject.Description) (valueobject.Identifier, error) {
 
+	// generate new uuid
 	id, _ := valueobject.NewIdentifier()
 
-	e := project.NewAggregate(id, shortCode, shortName, longName, description)
+	// initialize an array of existing short codes
+	var existingShortCodes []valueobject.ShortCode
+
+	// get a list of all the projects
+	existingProjects, _ := s.ListProjects(ctx, true)
+
+	// loop through each project and add each short code to the array of existing short codes
+	for _, proj := range existingProjects {
+		existingShortCodes = append(existingShortCodes, proj.ShortCode())
+	}
```

Comment:
From the performance optimisation point of view, shouldn't we have the list of shortcodes already kept somewhere instead of generating every time the new projects is created? So once it exists, only check against and add new one.

Reply:
I could be wrong, but I think this is just how event stores are supposed to work. We can't exactly store a list somewhere since we would just be storing the event and the list of short codes would need to be recreated anyways. I don't think the performance of this vs getting all the project data is any better albeit a bit overkill because I only need the short codes. If performance does become an issue, there is a way to cache things but I would have to look into it.

Reply:
I'm indeed taking the future into consideration, where you will have much more than few projects to iterate through. But we can forget about it for now.